js/binance.js

```diff
@@ -4230,31 +4286,39 @@ module.exports = class binance extends Exchange {
          * @param {object} params extra parameters specific to the binance api endpoint
          * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html#trade-structure}
          */
-        if (symbol === undefined) {
-            throw new ArgumentsRequired (this.id + ' fetchMyTrades() requires a symbol argument');
-        }
         await this.loadMarkets ();
-        const market = this.market (symbol);
-        const request = {
-            'symbol': market['id'],
-        };
-        const type = this.safeString (params, 'type', market['type']);
-        params = this.omit (params, 'type');
+        const request = {};
+        let market = undefined;
+        let type = undefined;
         let method = undefined;
         let marginMode = undefined;
-        [ marginMode, params ] = this.handleMarginModeAndParams ('fetchMyTrades', params);
-        if (type === 'spot' || type === 'margin') {
-            method = 'privateGetMyTrades';
-            if ((type === 'margin') || (marginMode !== undefined)) {
-                method = 'sapiGetMarginMyTrades';
-                if (marginMode === 'isolated') {
-                    request['isIsolated'] = true;
+        [ type, params ] = this.handleMarketTypeAndParams ('fetchMyTrades', undefined, params);
+        if (type === 'option') {
+            if (symbol !== undefined) {
+                const optionSeparateBase = symbol.split ('/');
+                const optionParts = symbol.split ('-');
```

Comment:
@Dan-krm why we can't use the market id here?

Reply:
Once the expiration date is passed for an option, there is no longer a market structure for it, so for methods that retrieve historical data and use a symbol argument it looks like we need to use a different approach. I think we can just have special handling in the individual methods that require it like I did here, but we could open it up for discussion and see if there is another approach. Let me know if there is a better way to go about this that I overlooked

Reply:
We might have this issue with delivery futures as well

Reply:
@Dan-krm I see, but what if the user provides the `marketId` directly? Shouldn't we handle that?